crates/router/src/connector/authorizedotnet/transformers.rs

```diff
@@ -133,12 +133,21 @@ fn get_pm_and_subsequent_auth_detail(
                     });
                     Ok((payment_details, processing_options, subseuent_auth_info))
                 }
-                _ => Err(errors::ConnectorError::NotSupported {
-                    message: format!("{:?}", item.request.payment_method_data),
-                    connector: "AuthorizeDotNet",
-                    payment_experience: api_models::enums::PaymentExperience::RedirectToUrl
-                        .to_string(),
-                })?,
+                api::PaymentMethodData::Wallet(_)
+                | api::PaymentMethodData::PayLater(_)
+                | api::PaymentMethodData::BankRedirect(_)
+                | api::PaymentMethodData::BankDebit(_)
+                | api::PaymentMethodData::BankTransfer(_)
+                | api::PaymentMethodData::Crypto(_)
+                | api::PaymentMethodData::MandatePayment
+                | api::PaymentMethodData::Reward(_)
+                | api::PaymentMethodData::Upi(_)
+                | api::PaymentMethodData::Voucher(_)
+                | api::PaymentMethodData::GiftCard(_) => {
+                    Err(errors::ConnectorError::NotImplemented(
+                        utils::get_unimplemented_payment_method_error_message("authorizedotnet"),
```

Comment:
Should we consider storing the connector name in a module level `const`, say `CONNECTOR_NAME`, so that all error messages or functions that need the connector name can use it instead?

Something like:

```rust
// This would go in the same file (`transformers.rs`), or in the connector-specific file (`authorizedotnet.rs`)
const CONNECTOR_NAME: &str = "authorizedotnet";

// Then the error propagation code would look like:
utils::get_unimplemented_payment_method_error_message(CONNECTOR_NAME)
```

I'm also fine with obtaining the string off of the `Display` implementation of an enum if that's available.

CC: @ArjunKarthik

Reply:
@SanchithHegde Agree, That would be a neat solution